src/components/Modal/Modal.tsx

```diff
@@ -64,178 +64,156 @@ export interface ModalProps extends FooterProps {
 }
 type CombinedProps = ModalProps & WithAppProviderProps;
 
-interface State {
-  iframeHeight: number;
-}
-
 const getUniqueID = createUniqueIDFactory('modal-header');
 
-class Modal extends React.Component<CombinedProps, State> {
-  static Section = Section;
-
-  state: State = {
-    iframeHeight: IFRAME_LOADING_HEIGHT,
-  };
-
-  private headerId = getUniqueID();
-  private activatorRef = React.createRef<HTMLDivElement>();
-
-  render() {
-    const {
-      children,
-      title,
-      src,
-      iFrameName,
-      open,
-      instant,
-      sectioned,
-      loading,
-      large,
-      limitHeight,
-      footer,
-      primaryAction,
-      secondaryActions,
-      polaris: {intl},
-      onScrolledToBottom,
-      activator,
-      onClose,
-    } = this.props;
-
-    const {iframeHeight} = this.state;
-
-    const iframeTitle = intl.translate('Polaris.Modal.iFrameTitle');
-
-    let dialog: React.ReactNode;
-    let backdrop: React.ReactNode;
-    if (open) {
-      const footerMarkup =
-        !footer && !primaryAction && !secondaryActions ? null : (
-          <Footer
-            primaryAction={primaryAction}
-            secondaryActions={secondaryActions}
-          >
-            {footer}
-          </Footer>
-        );
-
-      const content = sectioned
-        ? wrapWithComponent(children, Section, {})
-        : children;
-
-      const body = loading ? (
-        <div className={styles.Spinner}>
-          <Spinner />
-        </div>
-      ) : (
-        content
-      );
-
-      const bodyMarkup = src ? (
-        <iframe
-          name={iFrameName}
-          title={iframeTitle}
-          src={src}
-          className={styles.IFrame}
-          onLoad={this.handleIFrameLoad}
-          style={{height: `${iframeHeight}px`}}
-        />
-      ) : (
-        <Scrollable
-          shadow
-          className={styles.Body}
-          onScrolledToBottom={onScrolledToBottom}
+const Modal: React.FunctionComponent<CombinedProps> & {
+  Section: typeof Section;
+} = function Modal({
+  children,
+  title,
+  src,
+  iFrameName,
+  open,
+  instant,
+  sectioned,
+  loading,
+  large,
+  limitHeight,
+  footer,
+  primaryAction,
+  secondaryActions,
+  polaris: {intl},
+  onScrolledToBottom,
+  activator,
+  onClose,
+  onIFrameLoad,
+  onTransitionEnd,
+}: CombinedProps) {
+  const [iframeHeight, setIframeHeight] = useState(IFRAME_LOADING_HEIGHT);
+
+  const headerId = getUniqueID();
```

Comment:
getUniqueID/createUniqueIdFactory is  problematic as it's not stable between rerenders (i.e. every time you rerender the component then you get a new Id which causes pointless changes to the DOM). Use the useUniqueId hook instead. See https://github.com/Shopify/polaris-react/pull/2079 for more info

```suggestion
  const headerId = useUniqueId('modal-header');
```